src/rust/engine/engine_cffi/src/lib.rs

```diff
@@ -434,6 +440,12 @@ fn make_core(
       )
     })?;
 
+  let local_python_distribution_absolute_path = PathBuf::from(
+    local_python_distribution_absolute_path_buf
+        .to_string()
```

Comment:
`to_os_string()` is infallible, and `PathBuf` can be constructed straight from one.

Maybe we should even add a `to_pathbuf` function to `Buffer`?